### PR TITLE
HPCC-22925 Add LDAP Feature Resource WsCodeSignAccess

### PR DESCRIPTION
**Pull request by RussWhitehead**
Add a resource called WsCodeSignAccess to the configuration, allowing it to
be displayed in configuration manager for WsCodeSign service and in ECLWatch
permissions page. The WsCodeSign service, when implemented, will check this
resource in order to validate code signing priviledges

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@mayx Please review

**Review by mayx — APPROVED** [review]
Looks good.

**Comment by RussWhitehead** [comment]
@richardkchapman Please merge

**Comment by richardkchapman** [comment]
Some questions just because I don't know how LDAP works...

Does there need to be any change to the code that sets up an LDAP system to create the value?

Does access default to denied for a new resource like this?

File: testing/regress/environment.xml.in [hunk 1]
```diff
@@ -935,6 +944,10 @@
                           path="WsELKAccess"
                           resource="WsELKAccess"
                           service="ws_elk"/>
+    <AuthenticateFeature  description="Access to WS code signing service"
```
Comment (richardkchapman) [inline testing/regress/environment.xml.in]: Here (and everywhere else) I'm not sure the description is quite right - the flag affects (or will affect) other places besides the code signing service, and should say something like "Access to sign ECL code"

The resource name could also be changed to "CodeSignAccess"

**Comment by richardkchapman** [comment]
@RussWhitehead Some minor comments/questions. Also, is there an environment.xml.in in the LN repo that needs updating too?

File: testing/regress/environment.xml.in (hunk 1 above)
Comment (RussWhitehead) [reply testing/regress/environment.xml.in]: Changed description and renamed resource

**Comment by RussWhitehead** [comment]
@richardkchapman These named resources are automagically created in LDAP when the service is loaded, using whatever is the "default permissions" that have been set by the Admins. Since you were going to ask, our production SDS servers have set the default permissions as Allow Access/Read/Write for Administrators, and Allow Access/Read/Write/Full for Authenticated Users

**Comment by RussWhitehead** [comment]
@richardkchapman Please rereview. I will address the LN repo with separate PR

**Pull request merged**